### PR TITLE
Add background color to focused spot-list-items

### PR DESCRIPTION
**Pull request by b12f**
Closes https://community.openproject.org/projects/openproject/work_packages/43482/activity
Closes https://community.openproject.org/projects/openproject/work_packages/43544/activity
Closes https://community.openproject.org/projects/openproject/work_packages/43545/activity

### CODE REVIEW

**Comment by Kharonus** [comment]
Tested it, looks quite fine in the project selector. Just the question: "focusing" here means focus with our arrow key selection? ❓ Because the browser "focus" does not change background color.

I have some more questions though regarding this menu behaviour, even with those not blocking this very PR (maybe lets mention them to product team in the meeting?). This was just seen while testing it.

image

1) When I have a search and a nested project matches, the preselection is on the (not selectable) parent.

2) If I use tab to switch focus, the focus outline does not affect the "background selected" item, hence I get a state of two different "selected" items.

I think, those artifacts are shown by trying to put too much convenience into the component (with having the arrow key navigation while focusing the input, selection with enter while focusing the input, etc.). We should be clear, what drawbacks we are ready to take for convenience/consistency.

File: frontend/src/app/shared/components/header-project-select/list/header-project-select-list.component.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  Input,
+  Output,
+} from '@angular/core';
+import { CurrentProjectService } from 'core-app/core/current-project/current-project.service';
+import { SearchableProjectListService } from 'core-app/shared/components/searchable-project-list/searchable-project-list.service';
+import { IProjectData } from 'core-app/shared/components/searchable-project-list/project-data';
+import { PathHelperService } from 'core-app/core/path-helper/path-helper.service';
+
+@Component({
+  selector: '[op-header-project-select-list]',
```
Comment (Kharonus) [inline frontend/src/app/shared/components/header-project-select/list/header-project-select-list.component.ts:16]: can we disable this linter rule? We already agreed using components as attributes is fine for us.

File: frontend/src/app/shared/components/header-project-select/header-project-select.component.html [hunk 2]
```diff
@@ -46,24 +47,21 @@
       </spot-text-field>
 
       <ng-container *ngIf="(loading$ | async) === false; else loadingTemplate">
-        <ng-container *ngIf="(projects$ | async) as projects">
-          <ul
-            *ngIf="projects.length > 0; else noResultsTemplate"
-            op-project-list
```
Comment (Kharonus) [inline frontend/src/app/shared/components/header-project-select/header-project-select.component.html:52]: I see, that you replaced it, but was `op-project-list` deleted? Or is it still used elsewhere?

Comment (Kharonus) [reply frontend/src/app/shared/components/header-project-select/header-project-select.component.html:52]: nvm: `op-project-list` was renamed to `op-project-include-list`

File: frontend/src/app/shared/components/project-include/project-include.component.html [hunk 3]
```diff
@@ -104,37 +104,9 @@ <h1 class="op-project-list-modal--title">{{ text.title }}</h1>
   </ng-container>
 
   <ng-template #loadingTemplate>
-    <op-content-loader
+    <op-loading-project-list
       class="op-project-list-modal--loading"
```
Comment (Kharonus) [inline frontend/src/app/shared/components/project-include/project-include.component.html:111]: Why do we need a css class? Does it change somehow the behaviour of the standard `op-loading-project-list`?

**Review by Kharonus — COMMENTED** [review]
From code perspective I have no major issues. There is no new bugs introduced, as far as I can tell. The rework seems legit.

Nevertheless there are the mentioned issues still open with the selection/focus behavior of both lists.

**Review by Kharonus — CHANGES_REQUESTED** [review]
Just noticed, that those issues about focus vs. arrow-key-selection are introduced by this very PR, hence we do need to fix those here. :/ If you need help in reproducing them, please contact me.

File: frontend/src/app/shared/components/project-include/project-include.component.html (hunk 3 above)
Comment (b12f) [reply frontend/src/app/shared/components/project-include/project-include.component.html:111]: `frontend/src/global_styles/content/_project_list_modal.sass` sets some styles here, yes.

**Review by oliverguenther — APPROVED** [review]
There's a lot of linting errors due to formatting (indent etc.), but I haven't found any functional or code style issues so merging this to get it deployed on community.

**Pull request merged**